readthedocs/proxito/views/serve.py

```diff
@@ -176,16 +176,25 @@ def get(self,
             )
             raise Http404('Invalid URL for project with versions')
 
-        # TODO: un-comment when ready to perform redirect here
-        # redirect_path, http_status = self.get_redirect(
-        #     final_project,
-        #     lang_slug,
-        #     version_slug,
-        #     filename,
-        #     request.path,
-        # )
-        # if redirect_path and http_status:
-        #     return self.get_redirect_response(request, redirect_path, http_status)
+        redirect_path, http_status = self.get_redirect(
+            project=final_project,
+            lang_slug=lang_slug,
+            version_slug=version_slug,
+            filename=filename,
+            full_path=request.path,
+            forced_only=True,
+        )
+        if redirect_path and http_status:
+            try:
+                return self.get_redirect_response(
+                    request=request,
+                    redirect_path=redirect_path,
+                    proxito_path=request.path,
+                    http_status=http_status,
+                )
+            except InfiniteRedirectException:
+                # Continue with our normal serve.
```

Comment:
Should we log something here?

Reply:
We already log that here https://github.com/readthedocs/readthedocs.org/blob/0acd5e4cfe2369792ba03338f81aa81c23bd5c66/readthedocs/proxito/views/mixins.py#L248-L254

but we can add another here to be more specific

Reply:
We should add `force=` attribute to that log line to have more context when debugging.